### PR TITLE
[SRE-127] Setup nix CI

### PR DESCRIPTION
**Pull request by craigem**
This PR exists to establish the scaffolding for adding `cardano-rosetta-server` to Hydra CI.

### CODE REVIEW

**Comment by disassembler** [comment]
Why would we put all the nix files inside the JavaScript application? This makes nix-build of a release archive blow up in all sorts of bad ways?

**Comment by rhyslbw** [comment]
@disassembler The intent was to scope the nix build config to the relevant directory of the repo given it's only packaging the Node.js app. I can appreciate the value of standardising outweighs this though

File: nix/cardano-rosetta-server.nix [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  src = stdenv.mkDerivation {
+    pname = "${packageJSON.name}-src";
+    version = packageJSON.version;
+    buildInputs = [ yarn nodejs ];
+    src = nix-inclusive ./../cardano-rosetta-server [
+      ../cardano-rosetta-server/yarn.lock
+      ../cardano-rosetta-server/.yarnrc
+      ../cardano-rosetta-server/package.json
+      ../cardano-rosetta-server/packages-cache
+      ../cardano-rosetta-server/tsconfig.json
```
Comment (rhyslbw) [inline nix/cardano-rosetta-server.nix:22]: Need to add 
```
../cardano-rosetta-server/tsconfig-dist.json
```

**Review by rhyslbw — CHANGES_REQUESTED** [review]

File: nix/cardano-rosetta-server.nix (hunk 1 above)
Comment (craigem) [reply nix/cardano-rosetta-server.nix:22]: Added.

File: nix/cardano-rosetta-server.nix [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    cd $out
+
+    yarn build
+    find . -name node_modules -type d -print0 | xargs -0 rm -rf
+    yarn --production --offline --frozen-lockfile --non-interactive
+
+    mkdir -p $out/bin
+    cat <<EOF > $out/bin/${packageJSON.name}
+    #!${runtimeShell}
+    exec ${nodejs}/bin/node $out/packages/server/dist/index.js
```
Comment (rhyslbw) [inline nix/cardano-rosetta-server.nix:53]:
```suggestion
    exec ${nodejs}/bin/node $out/dist/src/server/index.js
```

**Review by rhyslbw — CHANGES_REQUESTED** [review]

File: nix/cardano-rosetta-server.nix (hunk 2 above)
Comment (craigem) [reply nix/cardano-rosetta-server.nix:53]: Updated.

File: nix/cardano-rosetta-server.nix [hunk 3]
```diff
@@ -49,10 +49,10 @@ in stdenv.mkDerivation {
     yarn --production --offline --frozen-lockfile --non-interactive
 
     mkdir -p $out/bin
-    cat <<EOF > $out/bin/cardano-rosetta-server
+    cat <<EOF > $out/bin/${packageJSON.name}
```
Comment (rhyslbw) [inline nix/cardano-rosetta-server.nix:52]: :+1:

**Review by rhyslbw — APPROVED** [review]

**Pull request merged**